Error: Numeric PreRelease version must not contain leading zeroes "00043"

```
Starting OpenShift using openshift/origin:2a00043 ...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for openshift/origin:2a00043 image ... OK
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... OK
-- Checking type of volume mount ... 
   Using nsenter mounter for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ... 
   Using 127.0.0.1 as the server IP
-- Starting OpenShift container ... 
   Creating initial OpenShift configuration
FAIL
   Error: could not update OpenShift configuration
   Caused By:
     Error: Numeric PreRelease version must not contain leading zeroes "00043"
[ERROR] PID 28245: hack/test-end-to-end-docker.sh:61: `oc cluster up --server-loglevel=4 --version="${TAG}" --host-data-dir="${VOLUME_DIR}/etcd" --host-volumes-dir="${VOLUME_DIR}"` exited with status 1.
```
https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/1230/consoleFull#111513341858b6e51eb7608a5981914356

@csrwng

--- a/pkg/bootstrap/docker/openshift/helper.go
+++ b/pkg/bootstrap/docker/openshift/helper.go
@@ -669,15 +669,17 @@ func (h *Helper) ServerPrereleaseVersion() (semver.Version, error) {
 	if len(versionStr) == 0 {
 		return semver.Version{}, fmt.Errorf("did not find version in command output")
 	}
+	return parseOpenshiftVersion(versionStr)
+}
 
+func parseOpenshiftVersion(versionStr string) (semver.Version, error) {
 	// The OCP version may have > 4 parts to the version string,
 	// e.g. 3.5.1.1-prerelease, whereas Origin will be 3.5.1-prerelease,
 	// drop the 4th digit for OCP.
-	re := regexp.MustCompile("([0-9]+).([0-9]+).([0-9]+).([0-9]+)(.*)")
+	re := regexp.MustCompile("([0-9]+)\\.([0-9]+)\\.([0-9]+)\\.([0-9]+)(.*)")
 	versionStr = re.ReplaceAllString(versionStr, "${1}.${2}.${3}${5}")
 
-	version, err := semver.Parse(versionStr)
-	return version, err
+	return semver.Parse(versionStr)
 }
 
 func useDNSIP(version semver.Version) bool {

--- a/pkg/bootstrap/docker/openshift/helper_test.go
+++ b/pkg/bootstrap/docker/openshift/helper_test.go
@@ -1,0 +1,44 @@
+package openshift
+
+import (
+	"testing"
+)
+
+func TestParseOpenshiftVersion(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{
+			input:    "1.3.1",
+			expected: "1.3.1",
+		},
+		{
+			input:    "3.5.5.4",
+			expected: "3.5.5",
+		},
+		{
+			input:    "3.5.5.18-alpha.3",
+			expected: "3.5.5-alpha.3",
+		},
+		{
+			input:    "3.4.144.2-1",
+			expected: "3.4.144-1",
+		},
+		{
+			input:    "3.6.0-alpha.2+2a00043-774",
+			expected: "3.6.0-alpha.2+2a00043-774",
+		},
+	}
+
+	for _, test := range tests {
+		v, err := parseOpenshiftVersion(test.input)
+		if err != nil {
+			t.Errorf("unexpected error for %s: %v", test.input, err)
+			continue
+		}
+		if v.String() != test.expected {
+			t.Errorf("unexpected output. Got: %v, expected: %s", v, test.expected)
+		}
+	}
+}

